// @progress/telerik-reporting-schematics/src/tests/ng17.spec.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const schematics_1 = require("@angular-devkit/schematics");
const testing_1 = require("@angular-devkit/schematics/testing");
const path = require("path");
const fs = require("fs");
const collectionPath = path.join(__dirname, '../../collection.json');
const componentCss = "/src/app/component/my-report-viewer/my-report-viewer.component.css";
const componentHtml = "/src/app/component/my-report-viewer/my-report-viewer.component.html";
const componentSpecTs = "/src/app/component/my-report-viewer/my-report-viewer.component.spec.ts";
const componentTs = "/src/app/component/my-report-viewer/my-report-viewer.component.ts";
const snapshotHtml = "my-report-viewer.component.html.snapshot";
const snapshotSpecTs = "my-report-viewer.component.spec.ts.snapshot";
const snapshotTs = "my-report-viewer.component.ts.snapshot";
describe("Angular 17", () => {
    it('standalone module', () => __awaiter(void 0, void 0, void 0, function* () {
        const runner = new testing_1.SchematicTestRunner('schematics', collectionPath);
        const app = schematics_1.Tree.empty();
        app.create("angular.json", JSON.stringify(angular17JsonNoDefault, null, "  "));
        app.create("package.json", JSON.stringify(angular17PackageJson, null, "  "));
        const tree = yield runner.runSchematic('native-report-viewer', {
            name: "MyReportViewer",
            path: "src/app/component",
            project: "ReportingApp"
        }, app);
        const expectedFiles = [
            "/angular.json",
            "/package.json",
            componentCss,
            componentHtml,
            componentSpecTs,
            componentTs
        ];
        expect(tree.files).toEqual(jasmine.arrayWithExactContents(expectedFiles));
        expect(tree.readJson("angular.json")).toEqual(angular17JsonWithDefaultTheme);
        expect(tree.readJson("package.json")).toEqual(angular17WithReportingPackageJson);
        expect(tree.readText(componentCss)).toEqual("");
        // Enable to generate expected results
        // fs.writeFileSync(path.join(__dirname, 'data/standalone/ng17', snapshotHtml), tree.readText(componentHtml), 'utf8');
        // fs.writeFileSync(path.join(__dirname, 'data/standalone/ng17', snapshotSpecTs), tree.readText(componentSpecTs), 'utf8');
        // fs.writeFileSync(path.join(__dirname, 'data/standalone/ng17', snapshotTs), tree.readText(componentTs), 'utf8');
        expect(tree.readText(componentHtml)).toEqual(fs.readFileSync(path.join(__dirname, 'data/standalone/ng17', snapshotHtml), 'utf8'));
        expect(tree.readText(componentSpecTs)).toEqual(fs.readFileSync(path.join(__dirname, 'data/standalone/ng17', snapshotSpecTs), 'utf8'));
        expect(tree.readText(componentTs)).toEqual(fs.readFileSync(path.join(__dirname, 'data/standalone/ng17', snapshotTs), 'utf8'));
        expect(runner.tasks).toEqual([
            {
                name: 'node-package',
                options: {
                    command: 'install',
                    quiet: true,
                    hideOutput: true,
                    workingDirectory: undefined,
                    packageManager: undefined,
                    packageName: undefined,
                    allowScripts: false
                }
            }, {
                name: 'run-schematic',
                options: { collection: '@angular/localize', name: 'ng-add', options: {} }
            }
        ]);
    }));
});
const angular17PackageJson = {
    "name": "angular17-app",
    "version": "0.0.0",
    "scripts": {
        "ng": "ng",
        "start": "ng serve",
        "build": "ng build",
        "watch": "ng build --watch --configuration development",
        "test": "ng test"
    },
    "private": true,
    "dependencies": {
        "@angular/animations": "^17.0.0",
        "@angular/common": "^17.0.0",
        "@angular/compiler": "^17.0.0",
        "@angular/core": "^17.0.0",
        "@angular/forms": "^17.0.0",
        "@angular/platform-browser": "^17.0.0",
        "@angular/platform-browser-dynamic": "^17.0.0",
        "@angular/router": "^17.0.0",
        "rxjs": "~7.8.0",
        "tslib": "^2.3.0",
        "zone.js": "~0.14.2"
    },
    "devDependencies": {
        "@angular-devkit/build-angular": "^17.0.0",
        "@angular/cli": "^17.0.0",
        "@angular/compiler-cli": "^17.0.0",
        "@types/jasmine": "~5.1.0",
        "jasmine-core": "~5.1.0",
        "karma": "~6.4.0",
        "karma-chrome-launcher": "~3.2.0",
        "karma-coverage": "~2.2.0",
        "karma-jasmine": "~5.1.0",
        "karma-jasmine-html-reporter": "~2.1.0",
        "typescript": "~5.2.2"
    }
};
const angular17JsonNoDefault = {
    "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
    "version": 1,
    "newProjectRoot": "projects",
    "projects": {
        "ReportingApp": {
            "projectType": "application",
            "schematics": {
                "@schematics/angular:component": {
                    "style": "scss"
                }
            },
            "root": "",
            "sourceRoot": "src",
            "prefix": "app",
            "architect": {
                "build": {
                    "builder": "@angular-devkit/build-angular:application",
                    "options": {
                        "outputPath": "dist/angular17-app",
                        "index": "src/index.html",
                        "browser": "src/main.ts",
                        "polyfills": [
                            "zone.js"
                        ],
                        "tsConfig": "tsconfig.app.json",
                        "inlineStyleLanguage": "scss",
                        "assets": [
                            "src/favicon.ico",
                            "src/assets"
                        ],
                        "styles": [
                            "src/styles.scss"
                        ],
                        "scripts": []
                    },
                    "configurations": {
                        "production": {
                            "budgets": [
                                {
                                    "type": "initial",
                                    "maximumWarning": "500kb",
                                    "maximumError": "1mb"
                                },
                                {
                                    "type": "anyComponentStyle",
                                    "maximumWarning": "2kb",
                                    "maximumError": "4kb"
                                }
                            ],
                            "outputHashing": "all"
                        },
                        "development": {
                            "optimization": false,
                            "extractLicenses": false,
                            "sourceMap": true
                        }
                    },
                    "defaultConfiguration": "production"
                },
                "serve": {
                    "builder": "@angular-devkit/build-angular:dev-server",
                    "configurations": {
                        "production": {
                            "buildTarget": "Angular17App:build:production"
                        },
                        "development": {
                            "buildTarget": "Angular17App:build:development"
                        }
                    },
                    "defaultConfiguration": "development"
                },
                "extract-i18n": {
                    "builder": "@angular-devkit/build-angular:extract-i18n",
                    "options": {
                        "buildTarget": "Angular17App:build"
                    }
                },
                "test": {
                    "builder": "@angular-devkit/build-angular:karma",
                    "options": {
                        "polyfills": [
                            "zone.js",
                            "zone.js/testing"
                        ],
                        "tsConfig": "tsconfig.spec.json",
                        "inlineStyleLanguage": "scss",
                        "assets": [
                            "src/favicon.ico",
                            "src/assets"
                        ],
                        "styles": [
                            "src/styles.scss"
                        ],
                        "scripts": []
                    }
                }
            }
        }
    }
};
const angular17JsonWithDefaultTheme = {
    "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
    "version": 1,
    "newProjectRoot": "projects",
    "projects": {
        "ReportingApp": {
            "projectType": "application",
            "schematics": {
                "@schematics/angular:component": {
                    "style": "scss"
                }
            },
            "root": "",
            "sourceRoot": "src",
            "prefix": "app",
            "architect": {
                "build": {
                    "builder": "@angular-devkit/build-angular:application",
                    "options": {
                        "outputPath": "dist/angular17-app",
                        "index": "src/index.html",
                        "browser": "src/main.ts",
                        "polyfills": [
                            "zone.js"
                        ],
                        "tsConfig": "tsconfig.app.json",
                        "inlineStyleLanguage": "scss",
                        "assets": [
                            "src/favicon.ico",
                            "src/assets"
                        ],
                        "styles": [
                            "src/styles.scss",
                            "node_modules/@progress/kendo-theme-default/dist/default-ocean-blue.scss"
                        ],
                        "scripts": []
                    },
                    "configurations": {
                        "production": {
                            "budgets": [
                                {
                                    "type": "initial",
                                    "maximumWarning": "500kb",
                                    "maximumError": "1mb"
                                },
                                {
                                    "type": "anyComponentStyle",
                                    "maximumWarning": "2kb",
                                    "maximumError": "4kb"
                                }
                            ],
                            "outputHashing": "all"
                        },
                        "development": {
                            "optimization": false,
                            "extractLicenses": false,
                            "sourceMap": true
                        }
                    },
                    "defaultConfiguration": "production"
                },
                "serve": {
                    "builder": "@angular-devkit/build-angular:dev-server",
                    "configurations": {
                        "production": {
                            "buildTarget": "Angular17App:build:production"
                        },
                        "development": {
                            "buildTarget": "Angular17App:build:development"
                        }
                    },
                    "defaultConfiguration": "development"
                },
                "extract-i18n": {
                    "builder": "@angular-devkit/build-angular:extract-i18n",
                    "options": {
                        "buildTarget": "Angular17App:build"
                    }
                },
                "test": {
                    "builder": "@angular-devkit/build-angular:karma",
                    "options": {
                        "polyfills": [
                            "zone.js",
                            "zone.js/testing"
                        ],
                        "tsConfig": "tsconfig.spec.json",
                        "inlineStyleLanguage": "scss",
                        "assets": [
                            "src/favicon.ico",
                            "src/assets"
                        ],
                        "styles": [
                            "src/styles.scss"
                        ],
                        "scripts": []
                    }
                }
            }
        }
    }
};
const angular17WithReportingPackageJson = {
    "name": "angular17-app",
    "version": "0.0.0",
    "scripts": {
        "ng": "ng",
        "start": "ng serve",
        "build": "ng build",
        "watch": "ng build --watch --configuration development",
        "test": "ng test"
    },
    "private": true,
    "dependencies": {
        "@angular/animations": "^17.0.0",
        "@angular/common": "^17.0.0",
        "@angular/compiler": "^17.0.0",
        "@angular/core": "^17.0.0",
        "@angular/forms": "^17.0.0",
        "@angular/platform-browser": "^17.0.0",
        "@angular/platform-browser-dynamic": "^17.0.0",
        "@angular/router": "^17.0.0",
        "rxjs": "~7.8.0",
        "tslib": "^2.3.0",
        "zone.js": "~0.14.2"
    },
    "devDependencies": {
        "@angular-devkit/build-angular": "^17.0.0",
        "@angular/cli": "^17.0.0",
        "@angular/compiler-cli": "^17.0.0",
        "@angular/localize": "^17.0.0",
        "@progress/kendo-theme-default": "~7.0.0",
        "@progress/telerik-angular-native-report-viewer": "*",
        "@types/jasmine": "~5.1.0",
        "jasmine-core": "~5.1.0",
        "karma": "~6.4.0",
        "karma-chrome-launcher": "~3.2.0",
        "karma-coverage": "~2.2.0",
        "karma-jasmine": "~5.1.0",
        "karma-jasmine-html-reporter": "~2.1.0",
        "typescript": "~5.2.2"
    }
};
